src/System.Linq/src/System/Linq/Single.cs

```diff
@@ -133,27 +117,79 @@ public static TSource SingleOrDefault<TSource>(this IEnumerable<TSource> source,
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.predicate);
             }
 
-            using (IEnumerator<TSource> e = source.GetEnumerator())
+            if (!TryGetSingle(source, predicate, out TSource result))
+            {
+                return default(TSource);
+            }
+
+            return result;
+        }
+
+        private static bool TryGetSingle<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate, out TSource result)
+        {
+            if (source == null)
```

Comment:
These argument checks could be removed here, or from the calling methods. Or not at all, any preference?

Reply:
Moved forward on this and removed the argument checks from the calling methods, in line with `First.cs`